src/main/java/sysc4806/project24/mini_survey_monkey/controllers/CreateController.java

```diff
@@ -92,4 +92,119 @@ public String deleteQuestion(
         questionRepository.deleteById(questionID);
         return "redirect:/create/" + surveyID;
     }
+
+    @PostMapping("/create/{surveyID}/question/add")
+    public String addQuestion(
+            @PathVariable("surveyID") int surveyID,
+            @RequestParam("type") String type,
+            @RequestParam(value = "questionText", required = false) String questionText,
+            @RequestParam(value = "options", required = false) List<String> options,
+            @RequestParam(value = "minValue", required = false) Integer minValue,
+            @RequestParam(value = "maxValue", required = false) Integer maxValue) {
+
+        Survey survey = surveyRepository.findById(surveyID);
+
+        if (survey == null) {
+            throw new IllegalArgumentException("Survey not found");
+        }
+
+        // Add question based on type
+        if ("multiple-choice".equals(type)) {
+            MultipleChoiceQuestion mcQuestion = new MultipleChoiceQuestion();
+            mcQuestion.setQuestion(questionText != null ? questionText : "New Multiple Choice Question");
+            mcQuestion.setOptions(options != null ? options : Arrays.asList("Option 1", "Option 2"));
+            survey.addQuestion(mcQuestion);
+        } else if ("scale".equals(type)) {
+            ScaleQuestion scaleQuestion = new ScaleQuestion();
+            scaleQuestion.setQuestion(questionText != null ? questionText : "New Scale Question");
+            scaleQuestion.setMinValue(minValue != null ? minValue : 1);
+            scaleQuestion.setMaxValue(maxValue != null ? maxValue : 10);
+            survey.addQuestion(scaleQuestion);
+        }
```

Comment:
I like this implementation! 

Much easier to just have all question adding at endpoint and just pass what question type as a param.